src/main/java/org/example/DAOs/TicketDAO.java

```diff
@@ -3,29 +3,42 @@
 import org.example.entities.Ticket;
 import org.example.exceptions.DAOException;
 import org.example.mappers.TicketMapper;
-import org.example.utilities.DatabaseConnection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Repository;
 
+import javax.sql.DataSource;
+import javax.transaction.Transactional;
 import java.sql.*;
 import java.util.ArrayList;
 import java.util.List;
 
+@Repository
 public class TicketDAO implements DAO<Ticket> {
 
-    private final Connection connection;
+    private final DataSource dataSource;
     private final TicketMapper ticketMapper;
 
-    public TicketDAO() {
-        this.connection = DatabaseConnection.getConnection();
-        this.ticketMapper = new TicketMapper();
+    @Autowired
+    @Lazy
```

Comment:
Why here @Lazy is used?)

Reply:
@Lazy is used to prevent circular dependencies by delaying the initialization of TicketDAO until it is actually needed. In this case, TicketMapper depends on ClientDAO, ClientDAO depends on TicketDAO, and TicketDAO depends on TicketMapper. Using @Lazy delays the initialization of TicketDAO, which helps avoid circular dependency issues